src/d_demoloop.c

```diff
@@ -0,0 +1,233 @@
+    { "CREDIT",   "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO2",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "HELP2",    "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO3",    "",         0,   TYPE_DEMO, WIPE_MELT },
+};
+
+// Ultimate Doom
+demoloop_entry_t demoloop_retail[] = {
+    { "TITLEPIC", "D_INTRO",  170, TYPE_ART,  WIPE_MELT },
+    { "DEMO1",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "CREDIT",   "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO2",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "CREDIT",   "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO3",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "DEMO4",    "",         0,   TYPE_DEMO, WIPE_MELT },
+};
+
+// Doom II & fixed Final Doom
+demoloop_entry_t demoloop_commercial[] = {
+    { "TITLEPIC", "D_DM2TTL", 385, TYPE_ART,  WIPE_MELT },
+    { "DEMO1",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "CREDIT",   "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO2",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "TITLEPIC", "D_DM2TTL", 385, TYPE_ART,  WIPE_MELT },
+    { "DEMO3",    "",         0,   TYPE_DEMO, WIPE_MELT },
+};
+
+// TNT: Evilution & The Plutonia Experiement
+demoloop_entry_t demoloop_final[] = {
+    { "TITLEPIC", "D_DM2TTL", 385, TYPE_ART,  WIPE_MELT },
+    { "DEMO1",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "CREDIT",   "",         200, TYPE_ART,  WIPE_MELT },
+    { "DEMO2",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "TITLEPIC", "D_DM2TTL", 385, TYPE_ART,  WIPE_MELT },
+    { "DEMO3",    "",         0,   TYPE_DEMO, WIPE_MELT },
+    { "DEMO4",    "",         0,   TYPE_DEMO, WIPE_MELT },
+};
+
+// For local parsing purposes only.
+static demoloop_entry_t current_entry;
+
+demoloop_t demoloop;
+int        demoloop_count = 0;
+int        demoloop_current = -1;
+
+
+// TODO: actually finish this and make it work
+demoloop_entry_t D_ParseDemoLoopEntry(json_t *entry)
+{
+    const char* primary_lump   = JS_GetStringValue(entry, "primarylump");
+    const char* secondary_lump = JS_GetStringValue(entry, "secondarylump");
+    double      seconds        = JS_GetNumberValue(entry, "duration");
+    int         type           = JS_GetIntegerValue(entry, "type");
+    int         outro_wipe     = JS_GetIntegerValue(entry, "outro_wipe");
+
+    // We don't want a malformed entry to creep in, and break the titlescreen.
+    // If one such entry does exist, skip it.
+    // TODO: modify later to check locally for lump type.
+    if (type <= TYPE_NONE || type > TYPE_DEMO)
+    {
+        return dl_none;
+    }
+
+    // Similarly, but this time it isn't game-breaking.
+    // Let it gracefully default to "closest vanilla behavior".
+    if (outro_wipe <= WIPE_NONE || outro_wipe > WIPE_MELT)
+    {
+        outro_wipe = WIPE_MELT;
+    }
+
+    // Providing the time in seconds is much more intuitive for the end users.
+    int duration = seconds * TICRATE;
+
+    demoloop_entry_t demoloop_entry = {
```

Comment:
Shouldn't this be static? You are returning a pointer to a struct that only exists in the scope of this function.